src/libopenrave-core/colladaparser/colladawriter.cpp

```diff
@@ -516,8 +516,29 @@ class ColladaWriter : public daeErrorHandler
         FOREACHC(itbody,listbodies) {
             BOOST_ASSERT((*itbody)->GetEnv()==_penv);
             BOOST_ASSERT(_mapBodyIds.find((*itbody)->GetEnvironmentId()) == _mapBodyIds.end());
-            _mapBodyIds[(*itbody)->GetEnvironmentId()] = globalid++;
-
+            const std::string& bodyId = (*itbody)->_id;
+            // try to restore what id was used
+            // check for id that has format bodyX_motion
+            if (bodyId.size() > 11 && bodyId.substr(0, 4) == "body" && bodyId.substr(bodyId.size()-7) == "_motion") {
+                try {
+                    int numericBodyId = boost::lexical_cast<int>(bodyId.substr(4, bodyId.size()-11));
+                    if (numericBodyId >= globalid) {
+                        globalid = numericBodyId+1;
+                    }
+                    _mapBodyIds[(*itbody)->GetEnvironmentId()] = numericBodyId;
```

Comment:
@ziyan how do you know if `numericBodyId` is unique? shouldn't you check in `_mapBodyIds`?

Reply:
hmm, I am assuming `_id` is unique, and `_mapBodyIds` at this point is empty, but I can check.